llvm/lib/Transforms/Instrumentation/IndirectCallPromotion.cpp

```diff
@@ -287,7 +290,18 @@ uint32_t IndirectCallPromoter::tryToPromote(
     NumOfPGOICallPromotion++;
     NumPromoted++;
   }
-  return NumPromoted;
+
+  const bool Changed = (NumPromoted != 0);
```

Comment:
Just return false early if NumPromoted is 0 (then Changed can be removed and just return true at the end further below).